cosmos/airflow/graph.py

```diff
@@ -409,6 +410,21 @@ def _get_dbt_dag_task_group_identifier(dag: DAG, task_group: TaskGroup | None) -
     return dag_task_group_identifier
 
 
+def calculate_detached_node_name(node: DbtNode, counter: list[int] = [0]) -> str:
```

Comment:
```suggestion
def calculate_detached_node_name(node: DbtNode, counter: list[int] | None = None) -> str:
```

Reply:
@pankajkoti this cannot be `None`, because the method relies on the signature to be able to be incremental

Reply:
thanks. Wrote my response in https://github.com/astronomer/astronomer-cosmos/pull/1464#discussion_r1916490212